src/library/scala/util/Random.scala

```diff
@@ -65,11 +75,23 @@ class Random(val self: java.util.Random) extends AnyRef with Serializable {
    */
   def nextInt(n: Int): Int = self.nextInt(n)
 
+  /** Returns a pseudorandom, uniformly distributed int value between min
+   *  (inclusive) and the specified value max (exclusive), drawn from this
+   *  random number generator's sequence.
+   */
+  def between(min: Int, max: Int): Int = min + self.nextInt(max - min)
+
   /** Returns the next pseudorandom, uniformly distributed long value
    *  from this random number generator's sequence.
    */
   def nextLong(): Long = self.nextLong()
 
+   /** Returns a pseudorandom, uniformly distributed long value between min
+   *  (inclusive) and the specified value max (exclusive), drawn from this
+   *  random number generator's sequence.
+   */
+  def between(min: Long, max: Long): Long = min + self.nextLong() % (max - min)
```

Comment:
You are missing something.  Try min = 0L, max = (1L << 62) + (1L << 61) and check how many values fall in the ranges 0..(1L << 61), (1L << 61)..(2L << 61), and (2L << 61)..(3L << 61).  Alternatively, think more carefully about what happens when the last bit of the range is wrapped.  Also, (max - min) may overflow.